examples/images_contours_and_fields/colormap_interactive_adjustment.py

```diff
@@ -3,90 +3,77 @@
 Interactive Adjustment of Colormap Range
 ========================================
 
-Demonstration of using colorbar, picker, and event functionality to make an
-interactively adjustable colorbar widget.
+Demonstration of how a colorbar can be used to interactively adjust the
+range of colormapping on an image. To use the interactive feature, you must
+be in either zoom or pan mode and click inside the colorbar.
 
-Left clicks and drags inside the colorbar axes adjust the high range of the
-color scheme. Likewise, right clicks and drags adjust the low range. The
-connected AxesImage immediately updates to reflect the change.
+When zooming, the bbox of the zoom region defines the new vmin and vmax of
+the norm. Right clicking while in zoom mode will expand the
+vmin and vmax proportionally to the selected region, in the same manner that
+one can zoom out of an axis. When panning, the vmin and vmax of the norm are
+both adjusted according to the direction of movement.
+
+The title on the axes show what happens if a zoom or pan occurs on the
+colorbar. All of the colorbars in the figure have interactivity enabled and
+only update the image that the colorbar is associated with.
 
 .. redirect-from:: /gallery/userdemo/colormap_interactive_adjustment
 """
-
-import numpy as np
 import matplotlib.pyplot as plt
-from matplotlib.backend_bases import MouseButton
-
-###############################################################################
-# Callback definitions
-
-
-def on_pick(event):
-    adjust_colorbar(event.mouseevent)
-
-
-def on_move(mouseevent):
-    if mouseevent.inaxes is colorbar.ax:
-        adjust_colorbar(mouseevent)
-
+import numpy as np
 
-def adjust_colorbar(mouseevent):
-    if mouseevent.button == MouseButton.LEFT:
-        colorbar.norm.vmax = max(mouseevent.ydata, colorbar.norm.vmin)
-    elif mouseevent.button == MouseButton.RIGHT:
-        colorbar.norm.vmin = min(mouseevent.ydata, colorbar.norm.vmax)
+start = 0
+stop = 2 * np.pi
+N = 1024
+t = np.linspace(start, stop, N)
+data1d = np.sin(t)
+
+data2d = np.sin(t[:, np.newaxis]) * np.cos(t[np.newaxis, :])
+
+zoom_axes = ["2d", "color zoom in", "color zoom out"]
+pan_axes = ["2d", "color pan up", "color pan down"]
+
+fig, ax_dict = plt.subplot_mosaic(
+    [
+        zoom_axes,
+        pan_axes
+    ],
+    constrained_layout=True,
+)
```

Comment:
I find the multitude of axes rather confusing. AFAICS they only differ in their initial clim values. I assume that's intended to give a good starting point for the respective interactions. However, all operation work reasonably well when starting on the "2d" image. It think it's better to have only the one image and let the user try all operations on that image.

I also found that panning on the image was really slow, likely because varying tick labels cause the constrained_layout solver to kick in again. One would have to deactivate constrained_layout after the first draw. But that's also obsolete if we can get away with one image.

Reply:
:+1: The reason for that was to demonstrate what the outcome of zooming/panning would do in each of the cases, but perhaps you're right it would be better to just let someone start interacting with the image and see for themselves. I updated this to have 3 images now with data ranges that are under/normal/over to try and get the sense that you can interact with the colorbar to identify new data regions.

Let me know what you think of this approach... Happy to go to just 1 image as well if you think that would be more clear.